kernel/src/snapshot.rs

```diff
@@ -933,8 +933,53 @@ mod tests {
         assert!(cp.is_none());
     }
 
-    fn valid_last_checkpoint() -> Vec<u8> {
-        r#"{"size":8,"sizeInBytes":21857,"version":1}"#.as_bytes().to_vec()
+    fn valid_last_checkpoint() -> (Vec<u8>, LastCheckpointHint) {
+        let expected = LastCheckpointHint {
+            version: 1,
+            size: 8,
+            parts: None,
+            size_in_bytes: Some(21857),
+            num_of_add_files: None,
+            checkpoint_schema: None,
+            checksum: None,
+            tags: None,
+        };
+        let data = r#"{"size":8,"sizeInBytes":21857,"version":1}"#.as_bytes().to_vec();
+        (data, expected)
+    }
+
+    fn valid_last_checkpoint_with_tags() -> (Vec<u8>, LastCheckpointHint) {
+        use std::collections::HashMap;
+
+        let (_, base_expected) = valid_last_checkpoint();
+
+        let mut expected_tags = HashMap::new();
+        expected_tags.insert(
+            "author".to_string(),
+            "test_read_table_with_last_checkpoint".to_string(),
+        );
+        expected_tags.insert("environment".to_string(), "snapshot_tests".to_string());
+        expected_tags.insert("created_by".to_string(), "delta-kernel-rs".to_string());
+
+        let expected = LastCheckpointHint {
+            tags: Some(expected_tags.clone()),
+            ..base_expected
+        };
+
+        // Build JSON from the expected struct
+        let data = format!(
+            r#"{{"size":{},"sizeInBytes":{},"version":{},"tags":{{{}}}}}"#,
```

Comment:
nit: avoid the business logic of formatting the tags and just put them in the string,  It seems like this would be less code?